tests/conftest.py

```diff
@@ -2621,3 +2621,19 @@ def ns_resources_and_connections_cleanup():
     request.addfinalizer(ns_resources_and_connections_cleanup)
 
     return _create_ns_resources
+
+
+def collect_logs_fixture(request):
+    """
+    This fixture collects ocs logs after tier execution and this will allow
+    to see the cluster's status after the execution on all execution status options.
+    """
+    def finalizer():
+        """
+        Tracking both logs separately reduce changes of collision
+        """
+        if not config.RUN['cli_params'].get('deploy') and not config.RUN['cli_params'].get('destroy'):
```

Comment:
This is the expected behavior because we collect the must-gather for any deployment status as well before.

Reply:
So I have two questions:

1. Where is this fixture being used?

2. What happens when we override the configuration and set `gather_on_deploy_success` to `False` yet we still want to gather logs for the tier executions. In that scenario, we won't gather logs for the tier execution if we combine the deployment and tier in one command.

My point here is that I don't think keying off of the `deploy`/`destroy` `cli_params` is the right way to determine if we should be retrieving must gather logs.

Reply:
I understand, might that be a better idea to search for the value provided for **test mark expression** ? 
Considering the following execution: ```run-ci tests/ -m tier1```
But in that case, I feel like I will have to search for keying off the deploy and the destroy again as the value of **-m**.
Am I missing something ?

Reply:
@clacroix12 I think no one does deployment and tier1 in one execution so we don't have to deal with that case I think at the moment.